RedHat-UX/red-hat-design-system
# feat(dialog): use native HTML dialog element

## What I did

1. Change internal structure of `rh-dialog` to use the native HTML `<dialog>` element.
2. Overhaul much of the CSS.
3. Change how we handle `#overlay` to use the CSS native `::backdrop` psuedo-element. 
4. Left notes about deprecating `#overlay`
5. Added the `accessible-label` property for when users want to manually label their modals + added docs.
6. Modal now has logic to decide when to use `accessible-label`, `aria-labelledby` + IDREF, and the `aria-label` of the trigger element.
7. Improved layout of modal contents when scrollbars are present
8. Increased the touch target of the close modal button.
9. Trap focus inside the modal for improved accessibility (WIP).

## Testing Instructions

1. Visit [the DP demo](https://deploy-preview-1865--red-hat-design-system.netlify.app/elements/dialog/demo/). 
2. Add all kinds of content to the modal header, body, and footer, see if you can break it.
3. See if focus trapping works with various different interactive elements (or none at all).

## To Do's

- [x] Fix Shift + Tab focus trapping
- [ ] Fix focus trap with Tab in video modal with YouTube embed
- [ ] Update/improve changeset
- [ ] Test across different browsers and AT
- [x] Look into why trigger/`rh-button` displays two buttons (one without a label) in the DP but not locally

## Notes to Reviewers

Some docs on the Guidelines page are incorrect as they have the modal body scrolling independently from the `header` and `footer`. That needs updated. 

WIP right now. Come back later for a more accurate description. 😇 

Closes #1242 and #1238.
